app/code/core/Mage/Persistent/Model/Session.php

```diff
@@ -120,9 +120,9 @@ protected function _beforeSave()
         if ($this->isObjectNew()) {
             $this->setWebsiteId(Mage::app()->getStore()->getWebsiteId());
             // Setting cookie key
-            do {
+            while (!$this->getResource()->isKeyAllowed($this->getKey())) {
```

Comment:
was it validated, that this is working as expected on the first run, when no key was yet set?

Reply:
Did not commit latest changes and opend issue instead ... https://github.com/magento-ecg/coding-standard/issues/70